src/main/java/com/cburch/logisim/fpga/data/IOComponentTypes.java

```diff
@@ -288,32 +436,111 @@ public static void paintPartialMap(Graphics2D g,
       case SevenSegment: hasDp = true;
       case SevenSegmentNoDp : {
         final var indexes = getSevenSegmentDisplayArray(hasDp);
-        final var partx = (width > height) ? (float) height / (float) 5.0 : (float) width / (float) 5.0;
-        final var party = (width > height) ? (float) width / (float) 7.0 : (float) height / (float) 7.0;
-        for (var xpos = 0; xpos < 5; xpos++) {
-          for (var ypos = 0; ypos < 7; ypos++) {
-            if (indexes[ypos][xpos] == pinNr) {
-              final var bx = (width > height) ? x + (int) ((float) ypos * party) : x + (int) ((float) xpos * partx);
-              final var by = (width > height) ? y + (int) ((float) xpos * partx) : y + (int) ((float) ypos * party);
-              final var bw = (width > height) ? x + (int) ((float) (ypos + 1) * party) - bx :
-                  x + (int) ((float) (xpos + 1) * partx) - bx;
-              final var bh = (width > height) ? y + (int) ((float) (xpos + 1) * partx) - by : 
-                  y + (int) ((float) (ypos + 1) * party) - by;
-              g.fillRect(bx, by, bw, bh);
+        var partX = (float) 0;
+        var partY = (float) 0;
+        switch (mapRotation) {
+          case rotationPlusNinety :
+          case rotationMinusNinety : {
+            partX = (float) width / (float) 7;
+            partY = (float) height / (float) 5; 
+            break;
+          }
+          default : {
+            partX = (float) width / (float) 5;
+            partY = (float) height / (float) 7; 
+            break;
+          }
+        }
+        var realXIndex = 0;
+        var realXIndexPlusOne = 0;
+        var realYIndex = 0;
+        var realYIndexPlusOne = 0;
+        for (var xIndex = 0; xIndex < 5; xIndex++) {
+          for (var yIndex = 0; yIndex < 7; yIndex++) {
+            if (indexes[yIndex][xIndex] == pinNr) {
+              switch (mapRotation) {
+                case rotationPlusNinety : {
+                  realXIndex = yIndex;
+                  realXIndexPlusOne = yIndex  + 1;
+                  realYIndex = 4 - xIndex;
+                  realYIndexPlusOne = 5 - xIndex;
+                  break;
+                }
+                case rotationMinusNinety : {
+                  realXIndex = 6 - yIndex;
```

Comment:
I'd rename to `realIndexX` (and also `yIndex` => `indexY`) and others like `realIndexXPlusOne` etc.

Reply:
I think it is in the eye of the beholder :-)

Reply:
I am trying to reduce linter complains :) But that's not crucial